src/queries/rum-content-requests.sql

```diff
@@ -1,40 +1,306 @@
---- offset: 0
 --- domainkey: secret
-SELECT
-  day,
-  contentrequests,
-  hostname
-FROM (
+WITH all_raw_events AS (
   SELECT
+    id,
+    checkpoint,
+    source,
+    target,
+    weight,
+    url,
+    host,
     hostname,
-    ROW_NUMBER() OVER (ORDER BY hostname) AS resrow,
-    FORMAT_TIMESTAMP('%F', TIMESTAMP_TRUNC(time, DAY, @timezone)) AS day,
-    SUM(pageviews) AS contentrequests
+    user_agent,
+    time,
+    TIMESTAMP_TRUNC(time, DAY, 'UTC') AS trunc_time # events will be evaluated on day level
   FROM
-    helix_rum.PAGEVIEWS_V5(
-      '-', # url
-      CAST(@offset AS INT64), # offset
-      -1, # days to fetch
+    helix_rum.EVENTS_V5(
+      @url, # list of urls
+      CAST(-1 AS INT64), # offset
+      CAST(-1 AS INT64), # days to fetch
       @startdate, # start date
       @enddate, # end date
-      @timezone, # timezone
-      'nobot', # deviceclass
-      @domainkey # domainkey
+      'UTC', # timezone
+      'all', # deviceclass
+      @domainkey # domain key to prevent data sharing
     )
-  WHERE (
-    hostname NOT LIKE '%.hlx.live'
-    AND hostname NOT LIKE '%.hlx3.live'
+  WHERE
+    hostname != ''
+    AND NOT REGEXP_CONTAINS(hostname, r'^\d+\.\d+\.\d+\.\d+$') -- IP addresses
+    AND hostname NOT LIKE 'localhost%'
     AND hostname NOT LIKE '%.hlx.page'
     AND hostname NOT LIKE '%.hlx3.page'
-    AND hostname NOT LIKE 'localhost'
-    AND hostname NOT LIKE '127.0.0.1'
-  )
-  GROUP BY day, hostname
-  ORDER BY hostname, day
+    AND hostname NOT LIKE '%.hlx.live'
+    AND hostname NOT LIKE '%.helix3.dev'
+    AND hostname NOT LIKE '%.sharepoint.com'
+    AND hostname NOT LIKE '%.aem.page'
+    AND hostname NOT LIKE '%.aem.live'
+),
+
+get_top_host AS (
+  SELECT
+    hostname,
+    APPROX_TOP_COUNT(host, 1)[OFFSET(0)].value AS top_host
+  FROM all_raw_events
+  GROUP BY hostname
+),
+
+# IDs can repeat, so we group by hostname and day
+group_all_events_daily AS (
+  SELECT
+    e.id,
+    th.top_host,
+    e.hostname,
+    e.trunc_time,
+    e.weight,
+    # an ID marks a single HTML request
+    e.weight AS html_requests,
+    # a JSON request
+    COALESCE(
+      COUNTIF(
+        e.checkpoint = 'loadresource'
+        AND e.target NOT LIKE '%.html'
+      ),
+      0
+    )
+    * e.weight AS json_requests,
+    # request by bot
+    COALESCE(COUNTIF(e.user_agent LIKE 'bot%') > 0, false) AS is_bot_request,
+    # request with 404 error
+    COALESCE((COUNTIF(e.checkpoint = '404') > 0), false) AS is_404_request,
+    # request to excluded url (html requests)
+    COALESCE(
+      (
+        COUNTIF(
+          e.url LIKE '%/manifest.json'
+          OR e.url LIKE '%/libs/%'
+        )
+        > 0
+      ),
+      false
+    ) AS is_excluded_url_request,
+    # request to excluded source (json requests)
+    COALESCE(
+      (
+        COUNTIF(
+          e.source LIKE '%/api/qraphql/%'
+        )
+        > 0
+      ), false
+    ) AS is_excluded_source_request
+  FROM all_raw_events AS e
```

Comment:
```suggestion
  FROM all_raw_events AS events
```
I've been wondering what `e` would be